core/rendered_connection.ts

```diff
@@ -554,29 +550,10 @@ export class RenderedConnection extends Connection {
 
     const parentBlock = this.getSourceBlock();
     const childBlock = renderedChildConnection.getSourceBlock();
-    const parentRendered = parentBlock.rendered;
-    const childRendered = childBlock.rendered;
 
-    if (parentRendered) {
-      parentBlock.updateDisabled();
-    }
-    if (childRendered) {
-      childBlock.updateDisabled();
-    }
-    if (parentRendered && childRendered) {
-      if (
-        this.type === ConnectionType.NEXT_STATEMENT ||
-        this.type === ConnectionType.PREVIOUS_STATEMENT
-      ) {
-        // Child block may need to square off its corners if it is in a stack.
-        // Rendering a child will render its parent.
-        childBlock.queueRender();
-      } else {
```

Comment:
I'm curious why you got rid of this logic - just not enough of a performance boost to justify it? or was not rendering the child block in all cases a mistake?

Reply:
Not rendering the child in all cases was a mistake. Since we added custom rendering, value input blocks may also change shape depending on if they're connected to a parent or not.